pkg/controller/jobframework/reconciler.go

```diff
@@ -164,6 +182,32 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, err
 	}
 
+	// 1.1 If the workload is pending deletion, suspend the job if needed
+	// and drop the finalizer.
+	if wl != nil && !wl.DeletionTimestamp.IsZero() {
+		log.V(2).Info("The workload is marked for deletion")
+		if !job.IsSuspended() {
```

Comment:
I'm not sure we want this check, as it was a source of issue in a recently fixed bug: https://github.com/kubernetes-sigs/kueue/pull/1005